clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -50,6 +50,7 @@
 
     private final long refreshBackoffMs;
     private final long metadataExpireMs;
+    private final long metadataStaleMs;
```

Comment:
If `metadataStaleMs` is always `metadataExpireMs` * 2, do we need this separate variable? Also, it seems more reasonable to have the `metadataStaleMs` to take both `metadataExpireMs` and `refreshBackoffMs` into consideration. Otherwise the metadata easily become stale if user has a long `refreshBackoffMs`.